src/Simulation/Native/src/simulator/capi.cpp

```diff
@@ -35,20 +35,21 @@ extern "C"
         _In_ unsigned id,
         _In_ unsigned n,
         _In_reads_(n) int* b,
-        _In_reads_(n) unsigned* q)
+        _In_reads_(n) intptr_t* q)
     {
         std::vector<Gates::Basis> bv;
         bv.reserve(n);
         for (unsigned i = 0; i < n; ++i)
             bv.push_back(static_cast<Gates::Basis>(*(b + i)));
-        std::vector<unsigned> qv(q, q + n);
+        std::vector<logical_qubit_id> qv(
+            reinterpret_cast<logical_qubit_id*>(q), reinterpret_cast<logical_qubit_id*>(q + n));
```

Comment:
These kinds of `reinterpret_cast` invocations are sufficient to compile but do not actually work, because `logical_qubit_id` (which is just `unsigned`) and `intptr_t` are not the same size. To convert an array of one to the other, we'd need a helper function akin to what the QIR wrapper around the fullstate simulator uses to ensure the ids are copied and converted one at a time:
https://github.com/microsoft/qsharp-runtime/blob/e61934496d3dbc7763e0013eb7ccda34322bde63/src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp#L145-L154

Reply:
@bettinaheim This is the array copy you were hoping to avoid, but it would either have to happen in the runtime, or have to happen in the fullstate simulator, or we'd have to update the AVX kernels to also take `intptr_t` instead of their current types.